pom.xml

```diff
@@ -134,58 +140,63 @@
 				<plugin>
 					<groupId>org.apache.maven.plugins</groupId>
 					<artifactId>maven-enforcer-plugin</artifactId>
-					<version>1.4.1</version>
+					<version>3.4.1</version>
 				</plugin>
 				<plugin>
 					<groupId>org.codehaus.mojo</groupId>
 					<artifactId>build-helper-maven-plugin</artifactId>
-					<version>3.0.0</version>
+					<version>3.4.0</version>
 				</plugin>
 				<plugin>
 					<groupId>org.apache.maven.plugins</groupId>
 					<artifactId>maven-compiler-plugin</artifactId>
-					<version>3.8.0</version>
+					<version>3.11.0</version>
 					<configuration>
 						<!-- the minimum java version supported by socomo -->
-						<release>7</release>
+						<release>17</release>
```

Comment:
As explained in previous comment, I want to keep this at 7, so that Socomo can be executed with as old JDK as JDK 7. So the source code level in modules _socomo-core_ and _socomo-maven_ must be 7, and this is driven by the above setting. Note that this doesn't mean that socomo source code must be developed with this old JDK. You can use newer JDK in your IDE and for the project, but the source code level here should be kept low.